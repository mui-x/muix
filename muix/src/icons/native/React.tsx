import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)