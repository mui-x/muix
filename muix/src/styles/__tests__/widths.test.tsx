//

import React from 'react'

import { TEngine } from 'reactxx-typings'
import { atomizeRuleset, toClassNames, WIDTH } from "reactxx-styles"

import { initPlatform, dump } from "./init-platform.t"

describe("SHEETER $WIDTHS", () => {

  const doTest = (isWeb: boolean) => {
    beforeEach(() => initPlatform(isWeb))
    let ruleset

    it("01 empty", () => {
      ruleset = atomizeRuleset([
        { color: 'green' },
        WIDTH([640, 1024], null, {}, undefined),
      ])
      expect(ruleset).toMatchSnapshot()
    })

    it("02 simple", () => {
      ruleset = atomizeRuleset([
        WIDTH([640, 1024], { color: 'red' }),
      ])
      expect(ruleset).toMatchSnapshot()
    })

    describe("03 toClassNamesRuleset's width change", () => {
      beforeEach(() => initPlatform(isWeb))

      const rulesets = [
        WIDTH([0, 640], { color: 'red' }),
        WIDTH([640, 1024], { color: 'green' }),
        WIDTH(1024, { color: 'blue' }),
      ]
      !window.isWeb && it("01: native, 300", () =>
        //**WIDHT**
        dump(toClassNames({ $widths: { actWidth: 300 } }, ...rulesets))
      )
      !window.isWeb && it("02: native, 640", () =>
        //**WIDHT**
        dump(toClassNames({ $widths: { actWidth: 641 } }, ...rulesets))
      )
      !window.isWeb && it("03: native, 1024", () =>
        //**WIDHT**
        dump(toClassNames({ $widths: { actWidth: 1025 } }, ...rulesets))
      )
      !window.isWeb && it("04: native, undefined", () =>
        dump(toClassNames(undefined, ...rulesets))
      )
      window.isWeb && it("05: web", () =>
        dump(toClassNames(undefined, ...rulesets))
      )
    })

  }

  describe("## NATIVE ##", () => doTest(false))
  describe("## WEB ##", () => doTest(true))

})

