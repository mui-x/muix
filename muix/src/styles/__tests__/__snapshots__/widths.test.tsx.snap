// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHEETER $WIDTHS ## NATIVE ## 01 empty 1`] = `
Array [
  Object {
    "color": "green @[0]",
  },
]
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 02 simple 1`] = `
Array [
  Object {
    "color": "red @/$width",
  },
]
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 03 toClassNamesRuleset's width change 01: native, 300 1`] = `
Object {
  "color": Object {
    "tracePath": "/$width",
    "value": "red",
  },
}
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 03 toClassNamesRuleset's width change 01: native, 300 2`] = `
Object {
  "color": "red",
}
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 03 toClassNamesRuleset's width change 02: native, 640 1`] = `
Object {
  "color": Object {
    "tracePath": "/$width",
    "value": "green",
  },
}
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 03 toClassNamesRuleset's width change 02: native, 640 2`] = `
Object {
  "color": "green",
}
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 03 toClassNamesRuleset's width change 03: native, 1024 1`] = `
Object {
  "color": Object {
    "tracePath": "/$width",
    "value": "blue",
  },
}
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 03 toClassNamesRuleset's width change 03: native, 1024 2`] = `
Object {
  "color": "blue",
}
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 03 toClassNamesRuleset's width change 04: native, undefined 1`] = `undefined`;

exports[`SHEETER $WIDTHS ## NATIVE ## 03 toClassNamesRuleset's width change 04: native, undefined 2`] = `undefined`;

exports[`SHEETER $WIDTHS ## WEB ## 01 empty 1`] = `
Array [
  Array [
    ".a { color:green } @[0]",
  ],
]
`;

exports[`SHEETER $WIDTHS ## WEB ## 02 simple 1`] = `
Array [
  Array [
    "@media (min-width: 641px) and (max-width: 1024px).a { color:red } @/$width",
  ],
]
`;

exports[`SHEETER $WIDTHS ## WEB ## 03 toClassNamesRuleset's width change 01: native, 300 1`] = `
Array [
  "@media (min-width: 1025px).c { color:blue } @/$width",
  "@media (min-width: 641px) and (max-width: 1024px).b { color:green } @/$width",
  "@media (max-width: 640px).a { color:red } @/$width",
]
`;

exports[`SHEETER $WIDTHS ## WEB ## 03 toClassNamesRuleset's width change 01: native, 300 2`] = `"c b a"`;

exports[`SHEETER $WIDTHS ## WEB ## 03 toClassNamesRuleset's width change 02: native, 640 1`] = `
Array [
  "@media (min-width: 1025px).c { color:blue } @/$width",
  "@media (min-width: 641px) and (max-width: 1024px).b { color:green } @/$width",
  "@media (max-width: 640px).a { color:red } @/$width",
]
`;

exports[`SHEETER $WIDTHS ## WEB ## 03 toClassNamesRuleset's width change 02: native, 640 2`] = `"c b a"`;

exports[`SHEETER $WIDTHS ## WEB ## 03 toClassNamesRuleset's width change 03: native, 1024 1`] = `
Array [
  "@media (min-width: 1025px).c { color:blue } @/$width",
  "@media (min-width: 641px) and (max-width: 1024px).b { color:green } @/$width",
  "@media (max-width: 640px).a { color:red } @/$width",
]
`;

exports[`SHEETER $WIDTHS ## WEB ## 03 toClassNamesRuleset's width change 03: native, 1024 2`] = `"c b a"`;

exports[`SHEETER $WIDTHS ## WEB ## 03 toClassNamesRuleset's width change 04: native, undefined 1`] = `
Array [
  "@media (min-width: 1025px).c { color:blue } @/$width",
  "@media (min-width: 641px) and (max-width: 1024px).b { color:green } @/$width",
  "@media (max-width: 640px).a { color:red } @/$width",
]
`;

exports[`SHEETER $WIDTHS ## WEB ## 03 toClassNamesRuleset's width change 04: native, undefined 2`] = `"c b a"`;
